src/gui/tray/UserModel.cpp

```diff
@@ -617,11 +601,20 @@ void UserModel::addUser(AccountStatePtr &user, const bool &isCurrent)
     }
 
     if (!containsUser) {
-        beginInsertRows(QModelIndex(), rowCount(), rowCount());
-        _users << new User(user, isCurrent);
+        int row = rowCount();
+        beginInsertRows(QModelIndex(), row, row);
+
+        User * u = new User(user, isCurrent);
+
+        connect(u, &User::avatarChanged, this, [this, row]{
+           Q_EMIT dataChanged(index(row, 0), index(row, 0), {UserModel::AvatarRole});
```

Comment:
Q_EMIT is seldomly used, for consistency sake I'd favor emit

Reply:
IIRC lowercase keywords are disabled by default in Qt6, but for consistency's sake I'll go with emit

Reply:
Ah good to know. I guess there'll be a mass change somewhere in the future. :-)